react/RatingSummary.tsx

```diff
@@ -115,13 +181,53 @@ const RatingSummary: FunctionComponent<Props> = props => {
           args: { average },
         })
       })
+
+    client
+      .query({
+        query: AppSettings,
+      })
+      .then((response: ApolloQueryResult<SettingsData>) => {
+        const settings = response.data.appSettings
+        dispatch({
+          type: 'SET_SETTINGS',
+          args: { settings },
+        })
+      })
   }, [client, productId])
 
+  useEffect(() => {
+    window.__RENDER_8_SESSION__.sessionPromise.then((data: any) => {
+      const sessionRespose = data.response
+
+      if (!sessionRespose || !sessionRespose.namespaces) {
+        return
+      }
+
+      const { namespaces } = sessionRespose
+      const storeUserId = path(
+        ['authentication', 'storeUserId', 'value'],
+        namespaces
+      )
```

Comment:
You can avoid ramda with optional chaining in TS files
```suggestion
      const storeUserId = namespaces?.authentication?.storeUserId?.value
```